### PR TITLE
core: introduce a new DirectiveSequencer

### PR DESCRIPTION
**Pull request by webispy**
The DirectiveSequencer delivers the directives received from the
network to the CapabilityAgent according to the dialog-id and
blocking policy.

Each Capability agent must register blocking policies through
DirectPolicySequencer::addPolicy() to handle the Directive in
accordance with the policy.

In addition, listeners must be registered to receive directive
events from the DirectiveSequencer.

Signed-off-by: Inho Oh <inho.oh@sk.com>

### CODE REVIEW

File: include/clientkit/capability.hh [hunk 1]
```diff
@@ -180,10 +180,17 @@ public:
...
      * @param[in] ename event name
      * @param[in] dname directive name
-     * @return referred dialog request id
      */
     void addReferrerEvents(const std::string& ename, const std::string& dname);
 
+    /**
+     * @brief Add blocking policy for directive name
+     * @param[in] dname directive name
+     * @param[in] medium medium information
```
Comment (JeanTracker) [inline include/clientkit/capability.hh:189]: The `medium` should be removed in comment.

File: tests/core/test-core-directive-sequencer.cc [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+    nugu_directive_unref(ndir);
+
+    /* prehandle test */
+    counter2->value = 0;
+    ndir = directive_new("TTS", "test", "dlg1", "msg1");
+    seq.addListener("TTS", counter2);
+    g_assert(seq.add(ndir) == true);
+    g_assert(counter2->value == 1); /* onHandleDirective ignored */
+    seq.complete(ndir);
+    seq.removeListener("TTS", counter2);
```
Comment (JeanTracker) [inline tests/core/test-core-directive-sequencer.cc:137]: There is memory leak `ndir`. Please check the code below.

File: include/clientkit/capability.hh (hunk 1 above)
Comment (webispy) [reply include/clientkit/capability.hh]: Thank you :)

File: tests/core/test-core-directive-sequencer.cc (hunk 2 above)
Comment (webispy) [reply tests/core/test-core-directive-sequencer.cc:137]: The ndir is destroyed in the `seq.complete(ndir)`

File: src/core/capability_manager.cc [hunk 3]
```diff
@@ -62,6 +63,34 @@ void CapabilityManager::destroyInstance()
...
+
+bool CapabilityManager::onHandleDirective(NuguDirective* ndir)
+{
+    nugu_info("preprocessDirective");
+    preprocessDirective(ndir);
+
+    ICapabilityInterface* cap = findCapability(nugu_directive_peek_namespace(ndir));
+
+    nugu_info("processDirective");
+    cap->processDirective(ndir);
```
Comment (kimhyungrok) [inline src/core/capability_manager.cc:89]: Is it guaranteed that cap is never nullptr?

Comment (webispy) [reply src/core/capability_manager.cc:100]: Thank you. I will add nullptr check logic.

**Pull request merged**